lib/charms/layer/apache_zeppelin.py

```diff
@@ -168,3 +170,23 @@ def close_ports(self):
     def cleanup(self):
         self.dist_config.remove_dirs()
         unitdata.kv().set('zeppelin.installed', False)
+
+
+class ZeppelinAPI(object):
+    """
+    Helper for interacting with the Appache Zeppelin REST API.
+    """
+    def _url(self, *parts):
+        url = 'http://localhost:9090/api/'
+        for part in parts:
+            url = urljoin(url, part)
+        return url
+
+    def import_notebook(self, contents):
+        response = requests.post(self._url('notebook'), data=contents)
+        if response.status_code != 201:
+            return None
```

Comment:
The import specifically returns an accepted / rejected response to the requesting charm, so that the error can be surfaced closer to where the (presumable) cause is.  `raise_for_status` here would just cause the Zeppelin charm to go into an error state and become unusable.  We should avoid charm error states whenever possible, especially in the case where it may not be the charm's fault (e.g., if it gets passed an invalid notebook).